test/expr/unary/keypath/keypath.swift

```diff
@@ -893,6 +893,23 @@ struct SR_12290 {
   }
 }
 
+// SR-12745
+struct SR12745 {
+  var x : Int = 0
+}
+
+func testKeyPathHole(_ s: SR12745) {
+  s[\.x] = 10 // expected-error {{cannot infer contextual type for key path; consider adding an explicit root type}}
+  // expected-error@-1 {{value of type 'SR12745' has no subscripts; did you mean to use a key path subscript?}} {{5-5=keyPath: }}
+  _ = \.x // expected-error {{cannot infer contextual type for key path; consider adding an explicit root type}}
+  s[\SR12745.x] // expected-error {{value of type 'SR12745' has no subscripts; did you mean to use a key path subscript?}} {{5-5=keyPath: }}
+
+  let _ : AnyKeyPath = \.x // expected-error {{cannot infer contextual type for key path; consider adding an explicit root type}}
+  // non key path produce the general diagnostic
+  s[1] // expected-error {{value of type 'SR12745' has no subscripts}} {{none}}
```

Comment:
Not sure why we are checking the diagnostic for `s[1]` here... We already have several tests for "has no subscripts" elsewhere.

Reply:
The idea was just to show the general case in the same block since its really simple, but I can remove it no problem.

Reply:
Fixed!